lnbits/settings.py

```diff
@@ -1,92 +1,298 @@
+        env_file = ".env"
+        env_file_encoding = "utf-8"
+        case_sensitive = False
+        json_loads = list_parse_fallback
+
+
+settings = Settings()
+
+settings.lnbits_path = str(path.dirname(path.realpath(__file__)))
 
 try:
-    LNBITS_COMMIT = (
+    settings.lnbits_commit = (
         subprocess.check_output(
-            ["git", "-C", LNBITS_PATH, "rev-parse", "HEAD"], stderr=subprocess.DEVNULL
+            ["git", "-C", settings.lnbits_path, "rev-parse", "HEAD"],
+            stderr=subprocess.DEVNULL,
         )
         .strip()
         .decode("ascii")
     )
 except:
-    LNBITS_COMMIT = "unknown"
+    settings.lnbits_commit = "docker"
+
+
+if not settings.lnbits_admin_ui:
+    logger.debug(f"Enviroment Settings:")
+    for key, value in settings.dict(exclude_none=True).items():
+        logger.debug(f"{key}: {value}")
+
+
+def set_cli_settings(**kwargs):
+    for key, value in kwargs.items():
+        setattr(settings, key, value)
+
+
+async def check_admin_settings():
+    if settings.lnbits_admin_ui:
+        try:
+            ext_db = importlib.import_module(f"lnbits.extensions.admin").db
+        except:
+            logger.error("could not import module lnbits.extensions.admin database")
+            raise
+
+        async with ext_db.connect() as db:
+            try:
+                row = await db.fetchone("SELECT * FROM admin.settings")
+                if not row or len(row) == 0:
+                    logger.warning(
+                        "admin.settings empty. inserting new settings and creating admin account"
+                    )
+
+                    from lnbits.core.crud import create_account
+
+                    account = await create_account()
+                    settings.lnbits_admin_users.insert(0, account.id)
+                    keys = []
+                    values = ""
+                    for key, value in settings.dict(exclude_none=True).items():
+                        keys.append(key)
+                        if type(value) == list:
+                            joined = ",".join(value)
+                            values += f"'{joined}'"
+                        if type(value) == int or type(value) == float:
+                            values += str(value)
+                        if type(value) == bool:
+                            values += "true" if value else "false"
+                        if type(value) == str:
+                            value = value.replace("'", "")
+                            values += f"'{value}'"
+                        values += ","
+                    q = ", ".join(keys)
+                    v = values.rstrip(",")
+                    sql = f"INSERT INTO admin.settings ({q}) VALUES ({v})"
+                    await db.execute(sql)
+                    logger.warning(
+                        "initialized admin.settings from enviroment variables."
+                    )
+
+                    row = await db.fetchone("SELECT * FROM admin.settings")
```

Comment:
`admin.settings` CRUD operations should be part of `crud.py` in the `admin` extension (not of core)

Reply:
d2cd9726